# This file divides each of the ~3500 songs into ten-second splits and labels each split by ragam name.

In [ ]:
! pip install librosa

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
! pip install mir_eval

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.7/90.7 KB 5.7 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for mir_eval: filename=mir_eval-0.7-py3-none-any.whl size=100718 sha256=4e244a4b84ab188ac8593aaeb57224fa25dc1de42dafc18140c69eb51c7937a6
  Stored in directory: /root/.cache/pip/wheels/e9/f5/d5/eb3db1d056253da195208853842bce745a84b29f44cab59b6c
Successfully built mir_eval


In [ ]:
!pip install pydub librosa music21

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
import librosa
import librosa.display as dsp
from librosa import display as librosadisplay
from IPython.display import Audio
import matplotlib.pyplot as plt
import numpy as np
import time
import warnings
import soundfile as sf
warnings.filterwarnings('ignore')
import pandas as pd
import os

from IPython.display import Audio, Javascript
from scipy.io import wavfile
from pydub import AudioSegment

In [ ]:
EXPECTED_SAMPLE_RATE = 16000

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
selected_ragas = ['Ahiri', 'Abheri', 'amrutavarSiNi', 'Anandabhairavi', 'AndOLikA', 'Arabhi', 'asAvEri', 'aThANA', 'bahudAri', 'balahamsa', 'bauLi', 'bEgaDA', 'behAg', 'bhairavi', 'bhUpaLaM', 'bilahari', 'bindumAlini', 'bowLi', 'brundAvanasArangA', 'BrundAvani', 'cakravAkam', 'candrajyOti', 'cArukEsi', 'cenjuruTi', 'darbAr', 'dEsh', 'dEvagAndhAri', 'dEvamanOhari', 'dhanyAsi', 'dvijAvanti', 'gambhIranATai', 'garuDadhvani', 'gauDamalhAr', 'gauLa', 'gauLipantu', 'ghaNTA', 'hamIrkalyANi', 'hamsadhvani', 'hamsagamini', 'hamsanAdam', 'hamsAnandi', 'hanumatODi', 'harikAmbhOji', 'hEmavati', 'hindOLam', 'husEni', 'jaganmOhini', 'janaranjani', 'jOnpuri', 'kalyANavasantam', 'kalyANi', 'kamalamanOhari', 'kAmavardhani', 'kAmavardhini', 'kAmbhOji', 'kAmbOji', 'kAnaDA', 'kannaDa', 'kannaDagauLa', 'kApi', 'kApinArAyaNi', 'karaharapriya', 'karNaranjani', 'kathanakutUhalam', 'kEdAragauLa', 'kEdAram', 'khamAs', 'kharaharapriyA', 'kIravANi', 'kuntalavarALi', 'kurinji', 'lalitA', 'latAngi', 'madhuvanti', 'madhyamAvati', 'malahari', 'mALavi', 'malayamArutam', 'mAND', 'maNirangu', 'mAnji', 'mAyAmALavagauLa', 'mishrashivaranjani', 'misrashivaranjani', 'mOhanakalyANi', 'mOhanam', 'mukhAri', 'nAdanAmakriyA', 'nAgasvarAvaLi', 'naLinakAnti', 'nArAyaNagauLa', 'nArirItigauLa', 'nATa', 'naTabhairavi', 'nATakapriyA', 'nATTai', 'nATTakurinji', 'navarOj', 'nAyaki', 'nIlAmbari', 'pantuvarALi', 'paraju', 'paras', 'pashupatipriyA', 'punnAgavarALi', 'pUrNacandrikA', 'pUrvikalyANi', 'ranjani', 'ravicandrikA', 'rEvagupti', 'rEvati', 'rItigauLa', 'sahAnA', 'sALagabhairavi', 'sAma', 'sAramati', 'sAranga', 'saurAshtram', 'saurASTra', 'sAvEri', 'shankarAbharaNam', 'shaNmukhapriyA', 'shivaranjani', 'shrI', 'shrIranjani', 'shubhapantuvarALi', 'shuddhadhanyAsi', 'shuddhasAvEri', 'shyAmA', 'simhEndramadhyamam', 'sindubhairavi', 'sourAshTram', 'suraTi', 'tODi', 'vAcaspati', 'valaji', 'varALi', 'vasantA', 'yadukulakAmbhOji', 'yamunAkalyANi']

In [ ]:
selected_ragas_SET = {'Ahiri', 'Abheri', 'amrutavarSiNi', 'Anandabhairavi', 'AndOLikA', 'Arabhi', 'asAvEri', 'aThANA', 'bahudAri', 'balahamsa', 'bauLi', 'bEgaDA', 'behAg', 'bhairavi', 'bhUpaLaM', 'bilahari', 'bindumAlini', 'bowLi', 'brundAvanasArangA', 'BrundAvani', 'cakravAkam', 'candrajyOti', 'cArukEsi', 'cenjuruTi', 'darbAr', 'dEsh', 'dEvagAndhAri', 'dEvamanOhari', 'dhanyAsi', 'dvijAvanti', 'gambhIranATai', 'garuDadhvani', 'gauDamalhAr', 'gauLa', 'gauLipantu', 'ghaNTA', 'hamIrkalyANi', 'hamsadhvani', 'hamsagamini', 'hamsanAdam', 'hamsAnandi', 'hanumatODi', 'harikAmbhOji', 'hEmavati', 'hindOLam', 'husEni', 'jaganmOhini', 'janaranjani', 'jOnpuri', 'kalyANavasantam', 'kalyANi', 'kamalamanOhari', 'kAmavardhani', 'kAmavardhini', 'kAmbhOji', 'kAmbOji', 'kAnaDA', 'kannaDa', 'kannaDagauLa', 'kApi', 'kApinArAyaNi', 'karaharapriya', 'karNaranjani', 'kathanakutUhalam', 'kEdAragauLa', 'kEdAram', 'khamAs', 'kharaharapriyA', 'kIravANi', 'kuntalavarALi', 'kurinji', 'lalitA', 'latAngi', 'madhuvanti', 'madhyamAvati', 'malahari', 'mALavi', 'malayamArutam', 'mAND', 'maNirangu', 'mAnji', 'mAyAmALavagauLa', 'mishrashivaranjani', 'misrashivaranjani', 'mOhanakalyANi', 'mOhanam', 'mukhAri', 'nAdanAmakriyA', 'nAgasvarAvaLi', 'naLinakAnti', 'nArAyaNagauLa', 'nArirItigauLa', 'nATa', 'naTabhairavi', 'nATakapriyA', 'nATTai', 'nATTakurinji', 'navarOj', 'nAyaki', 'nIlAmbari', 'pantuvarALi', 'paraju', 'paras', 'pashupatipriyA', 'punnAgavarALi', 'pUrNacandrikA', 'pUrvikalyANi', 'ranjani', 'ravicandrikA', 'rEvagupti', 'rEvati', 'rItigauLa', 'sahAnA', 'sALagabhairavi', 'sAma', 'sAramati', 'sAranga', 'saurAshtram', 'saurASTra', 'sAvEri', 'shankarAbharaNam', 'shaNmukhapriyA', 'shivaranjani', 'shrIranjani', 'shubhapantuvarALi', 'shuddhadhanyAsi', 'shuddhasAvEri', 'simhEndramadhyamam', 'sindubhairavi', 'sourAshTram', 'suraTi', 'tODi', 'vAcaspati', 'valaji', 'varALi', 'vasantA', 'yadukulakAmbhOji', 'yamunAkalyANi'}

In [ ]:
#deleted shri from set
#deletd shyama from set

In [ ]:
numRagas = 0
for raga in selected_ragas_SET:
  numRagas += 1
print(numRagas)

138


In [ ]:
numRagas = 0
for raga in selected_ragas:
  numRagas += 1
print(numRagas)

140


In [ ]:
ragamDict = {}
for ragam in selected_ragas:
  ragamDict[ragam] = 0

In [ ]:
import pandas as pd
import os

In [ ]:
import re

In [ ]:
directory_path = '/content/drive/MyDrive/N240Project/SP_downloads'
directory_files = os.listdir(directory_path)

In [ ]:
string = "example-string.to.split"


In [ ]:
print(len(directory_files))

3561


In [ ]:
directory_files_copy = directory_files.copy()
print(len(directory_files_copy))

3561


In [ ]:
directory_files_copy = directory_files.copy()

## Breaks each song title into chunks and searches for a matching ragam name fro our bank of ~140 ragams.

In [ ]:
numFound = 0
for file in directory_files_copy:
    
    split_string = re.split(r'[-.()_ ]', file)

    numMatches = 0

    for string in split_string:
        for ragam in selected_ragas_SET:
            if string.lower() == ragam.lower():
                numMatches += 1
                
    if numMatches >0:
        numFound += 1
    
    else:
        print(split_string)
    
print(numFound)

['04', 'nAda', 'tanumanisham', 'shankaram', 'cittaranjani', 'mp3']
['09', 'nAda', 'tanumanisham', 'shankaram', 'cittaranjani', 'mp3']
['13', 'nI', 'nAma', 'rUpamulaku', 'mangaLa', 'sourAshtra', 'Adhi', 'thyAgarAja', '', '1', '', 'mp3']
['05', 'vinavE', 'O', 'manasa', 'vivarambuna', 'vivardhani', 'mp3']
['11', 'karuNa', 'jUDu', 'ninnu', 'nammina', 'shrI', 'mp3']
['08', 'shrI', 'kamalAmbikE', 'shivE', 'shrI', 'mp3']
['04', 'sItamma', 'mAyamma', 'shrI', 'mp3']
['04', 'shrI', 'kamalAmbikE', 'shivE', 'shrI', 'mp3']
['14', 'shri', 'abhayAmba', 'shri', 'Adhi', 'm', 'dhikShithar', 'mp3']
['05', 'dEvAdi', 'dEva', 'shrI', 'vAsudEva', 'sunAdavinOdini', 'mp3']
['01', 'sAmi', 'ninnE', 'kOri', 'VarNa', 'shri', 'Adhi', 'garbhapurivAsa', 'mp3']
['09', 'tillAnA', 'dhanAshrI', 'mp3']
['03', 'endarO', 'mahAnubhAvulu', 'shrI', '', '1', '', 'mp3']
['05', 'vara', 'nArada', 'nArAyaNa', 'vijayashrI', 'mp3']
['11', 'bhajarE', 'shrI', 'rAmam', 'vakuLA', 'bharaNam', 'chApu', 'Bhadrachala', 'Ramadas', 'mp3']
['05

In [ ]:
len(songListRagamDict)

3316

3316 is the number of songs that were recognizable as 1 distinct ragam, not shree or shyama.

In [ ]:
print(len(directory_files))

3561


### Process each song into a wav file

In [ ]:
def convert_audio_for_model(user_file, output_file='converted_audio_file.wav'):
  audio = AudioSegment.from_file(user_file)
  audio = audio.set_frame_rate(EXPECTED_SAMPLE_RATE).set_channels(1)
  audio.export(output_file, format="wav")
  return output_file

In [ ]:
converted_audio_file = convert_audio_for_model(uploaded_file_name)

In [ ]:
for ragam in selected_ragas:
  if ragam.lower() in uploaded_file_name.lower():
    print(ragam)
    uploaded_file_name = ragam

In [ ]:
sample_rate, data1 = wavfile.read(converted_audio_file, 'rb')

### Split each song into ten-second sub recording wave files, and label each split with the name of the ragam of the song from which the split is cut.

In [ ]:
#audio, sr = librosa.load(file_name)
def splitSong(sample_rate, data1, ragam):
# Get number of samples for 2 seconds; replace 2 by any number

  ragam = ragam.lower()

  buffer = 10 * sample_rate

  samples_total = len(data1) 
  samples_wrote = 0
  index = ragamDict[ragam] + 1
  numSplits = 0
  while samples_wrote < samples_total:

      #check if the buffer is not exceeding total samples 
      if buffer > (samples_total - samples_wrote):
          buffer = samples_total - samples_wrote

      block = data1[samples_wrote : (samples_wrote + buffer)]
      
      pathName = "/content/drive/MyDrive/N240Project/RagamSplits3/"
      out_filename = pathName + ragam.lower() + "/" + ragam + "_" + "split_" + str(index)

      # Write 10 second segment
      sf.write(out_filename, block, sample_rate, format = 'WAV')
      index += 1
      numSplits += 1
      samples_wrote += buffer

  ragamDict[ragam] += numSplits

### Inspect the songs not able to be identified as any ragam.

In [ ]:
unidentifiedRagaFiles = []

numNotFound = 0
for file in directory_files:


  ragamFound = False

  if ragamFound == False:

    for ragam in derived_ragas:
      if ragam.lower() in file.lower():
        fileRagam = ragam
        ragamFound = True

  if ragamFound == False:

    for ragam in parent_ragas:
      if ragam.lower() in file.lower():
        fileRagam = ragam
        ragamFound = True
    
  if ragamFound == False:
    numNotFound += 1
    print("file not found", file)
    unidentifiedRagaFiles.append(file)
    continue 

print("number not found is ", numNotFound)

file not found 09_tani.mp3
file not found 05_charana_kamalAlayathai_thiruppugazh-2.61-MB.mp3
file not found 01-valachi_vachchiyunnA-navarAgamAlike-Adhi-PSI.mp3
file not found 11-pibarE_rAmarasam-Ahir_bhairav-Adhi-sadhAshiva_brahmEndhra.mp3
file not found 13_nI_nAma_rUpamulaku_mangaLa_sourAshtra_Adhi_thyAgarAja.mp3
file not found 06-raghuvamsha_suDhAbuDhi-kadhana_kuthUhala-Adhi-PSI.mp3
file not found 14_yAdava_nI_bA_yadukula_nandana-hamsavinOdini-purandaradAsa.mp3
file not found 10_tani.mp3
file not found 05-ranga_pUra_vihAra-bRndAvanasAraGga_rUpaka_muttuswAmi-dhIkshitar.mp3
file not found 01-valachi_vachchiyunnA-navarAgamAlike-Adhi-PSI (1).mp3
file not found 09_evarani_nirNayincirirA_dEvAmrtavarshaNi_tyAgarAjai.mp3
file not found 03_Sankara_KamalmanOhari_madurai_somu.mp3
file not found 05_E_rAmuni_nammitinO_nE_vakuLAbharaNam_tyAgarAja.mp3
file not found 14_Ennakavi_neelamani_composer.mp3
file not found 16_nI_nAma_rUpamulaku_mangaLa_sourAshtra_Adhi_thyAgarAja.mp3
file not found 06-raghu

## Note that many of these unidentifiable songs do not have a recognizable ragam in their title.

In [ ]:
unidentifiedRagaFiles

['09_tani.mp3',
 '05_charana_kamalAlayathai_thiruppugazh-2.61-MB.mp3',
 '01-valachi_vachchiyunnA-navarAgamAlike-Adhi-PSI.mp3',
 '11-pibarE_rAmarasam-Ahir_bhairav-Adhi-sadhAshiva_brahmEndhra.mp3',
 '13_nI_nAma_rUpamulaku_mangaLa_sourAshtra_Adhi_thyAgarAja.mp3',
 '06-raghuvamsha_suDhAbuDhi-kadhana_kuthUhala-Adhi-PSI.mp3',
 '14_yAdava_nI_bA_yadukula_nandana-hamsavinOdini-purandaradAsa.mp3',
 '10_tani.mp3',
 '05-ranga_pUra_vihAra-bRndAvanasAraGga_rUpaka_muttuswAmi-dhIkshitar.mp3',
 '01-valachi_vachchiyunnA-navarAgamAlike-Adhi-PSI (1).mp3',
 '09_evarani_nirNayincirirA_dEvAmrtavarshaNi_tyAgarAjai.mp3',
 '03_Sankara_KamalmanOhari_madurai_somu.mp3',
 '05_E_rAmuni_nammitinO_nE_vakuLAbharaNam_tyAgarAja.mp3',
 '14_Ennakavi_neelamani_composer.mp3',
 '16_nI_nAma_rUpamulaku_mangaLa_sourAshtra_Adhi_thyAgarAja.mp3',
 '06-raghuvamsha_suDhAbuDhi-kadhana_kuthUhala-Adhi-PSI (1).mp3',
 '15_tillAna_dhirana_dhImta_kamAch_composer.mp3',
 '11-pibarE_rAmarasam-Ahir_bhairav-Adhi-sadhAshiva_brahmEndhra (1).mp3',


In [ ]:
#splits_directory_path = '/content/drive/MyDrive/N240Project/AudioSplits1'
#splits_directory_files = os.listdir(splits_directory_path)

## inspect our initialized dictionary counting # splits per ragam across all songs

In [ ]:
ragamDict

{'Ahiri': 0,
 'Abheri': 0,
 'amrutavarSiNi': 0,
 'Anandabhairavi': 0,
 'AndOLikA': 0,
 'Arabhi': 0,
 'asAvEri': 0,
 'aThANA': 0,
 'bahudAri': 0,
 'balahamsa': 0,
 'bauLi': 0,
 'bEgaDA': 0,
 'behAg': 0,
 'bhairavi': 0,
 'bhUpaLaM': 0,
 'bilahari': 0,
 'bindumAlini': 0,
 'bowLi': 0,
 'brundAvanasArangA': 0,
 'BrundAvani': 0,
 'cakravAkam': 0,
 'candrajyOti': 0,
 'cArukEsi': 0,
 'cenjuruTi': 0,
 'darbAr': 0,
 'dEsh': 0,
 'dEvagAndhAri': 0,
 'dEvamanOhari': 0,
 'dhanyAsi': 0,
 'dvijAvanti': 0,
 'gambhIranATai': 0,
 'garuDadhvani': 0,
 'gauDamalhAr': 0,
 'gauLa': 0,
 'gauLipantu': 0,
 'ghaNTA': 0,
 'hamIrkalyANi': 0,
 'hamsadhvani': 0,
 'hamsagamini': 0,
 'hamsanAdam': 0,
 'hamsAnandi': 0,
 'hanumatODi': 0,
 'harikAmbhOji': 0,
 'hEmavati': 0,
 'hindOLam': 0,
 'husEni': 0,
 'jaganmOhini': 0,
 'janaranjani': 0,
 'jOnpuri': 0,
 'kalyANavasantam': 0,
 'kalyANi': 0,
 'kamalamanOhari': 0,
 'kAmavardhani': 0,
 'kAmavardhini': 0,
 'kAmbhOji': 0,
 'kAmbOji': 0,
 'kAnaDA': 0,
 'kannaDa': 0,
 'kannaDa

In [ ]:
# for loop over all selected ragams
#os.mkdir('/content/malayalam_keyphrase_extraction')

counter = 0
for ragam in selected_ragas_SET:
    counter +=1
    newFolder = "/content/drive/MyDrive/N240Project/RagamSplits3/" + ragam.lower()
    print(counter)
    os.mkdir(newFolder)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138


In [ ]:
ragamDict = {}
for ragam in selected_ragas_SET:
  ragamDict[ragam.lower()] = 0

In [ ]:
songListRagamDict

{'03-dEvadEva_jagadIsvara-pUrvikalyANi.mp3': 'pUrvikalyANi',
 '05_ninnuvinAga_mari_dikkevarunnAru_Purvikalyani_Chapu.mp3': 'Purvikalyani',
 '03-nagumOmu_ganalEni-AbhEri (2).mp3': 'AbhEri',
 '03-paripUrNa_kAma_bhAvamuna-pUrvikalyANi.mp3': 'pUrvikalyANi',
 '02-tillai_cidambaram-pUrvikalyANi-gOpalakrishna_bharati.mp3': 'pUrvikalyANi',
 '01-mInAkShi_mEmudham_dhEhi-pUrvikalyANi-Adhi-m_dhIkShithar.mp3': 'pUrvikalyANi',
 '02-dEvadEva_jagadIsvara-pUrvikalyANi.mp3': 'pUrvikalyANi',
 '02-gangAdhAra_tripura-pUrvikalyANi.MP3': 'pUrvikalyANi',
 '03-mInAkSi_mEmudam_dEhi-pUrvikalyANi (1).mp3': 'pUrvikalyANi',
 '07-baNTurIti_koluviyavayya-hamsanAdam.mp3': 'hamsanAdam',
 '15_kshIrAbdhi_kanyaku_shrI_mahAlakshmikini_nira_Kurinji_ChApu_annamAchArya.mp3': 'Kurinji',
 '02_sharaNu_janakana_bilahari-Adi_tisra-Purandara_dhAsa.mp3': 'bilahari',
 '03-Ananda_naTam_AduvAr-pUrvikalyANi.mp3': 'pUrvikalyANi',
 '02-paripUrNa_kAma_bhAvamuna-pUrvikalyANi-rUpaka-thyAgarAja.mp3': 'pUrvikalyANi',
 '03-jnAnamosaga_rAdA-pUrv

## Update dictionary with split counts for each ragam

In [ ]:
filesCount = 0

for file in directory_files_copy:

  filesCount += 1  

  print(filesCount)
  print(file)

  """fileRagam = "noRagamFound"

  ragamFound = False

  if ragamFound == False:

    for ragam in derived_ragas:
      if ragam.lower() in file.lower():
        fileRagam = ragam
        ragamFound = True

  if ragamFound == False:

    for ragam in parent_ragas:
      if ragam.lower() in file.lower():
        fileRagam = ragam
        ragamFound = True
    
  if ragamFound == False:
    print("file not found", file)
    continue""" 

  fileDestination = "/content/drive/MyDrive/N240Project/SP_downloads/" + file

  converted_audio_file = convert_audio_for_model(fileDestination)

  sample_rate, data1 = wavfile.read(converted_audio_file, 'rb')

  fileRagam = songListRagamDict[file].lower()

  splitSong(sample_rate, data1, fileRagam)


print(ragamDict)




Streaming output truncated to the last 5000 lines.
NRK-05---Mamava-Meenakshi---Varali.mp3
818
13-Devaranama-Pantuvarali-Purandara.mp3
819
03-nE_pogaDakuNTE_nIkEmi-varALi.mp3
820
04_MamavaMeenakshi_Varali_MEka_MuttuswamiDixitar.mp3
821
04_kA_vA_vA_kandA_vA_vA_varALi_pApanAsam_sivanK.mp3
822
03_kAmAkshi-varAli-m.cApu_shyAmAsAstri.mp3
823
05-kAmAkSi_bangAru_kAmAkSi-varALi.mp3
824
03-mAmava_mInAkSi_rAjamAtangi-varALi-m_dIkshitar.mp3
825
04-marakata_lingam_cintayEham-vasantA.mp3
826
03-rAmacandram_bhAvayAmi-vasantA.mp3
827
02-sItamma_mAyamma-vasantA.mp3
828
03-brhAdambikAyai-vasantA.mp3
829
04-sItamma_mAyamma_shrI-vasanta.mp3
830
06-harihara_putram_shAstAram-vasantA.mp3
831
01-mAl_marugA_SaNmukhA-vasanta-pApanAsam_sivan.mp3
832
05-paramapuruSa_jagad-vasantA.mp3
833
07_sItamma_mAyamma_shrI_vasanta_tyAgarAja.mp3
834
05-naTanam_AdinAr_vegu-vasantA.mp3
835
02-sItamma_mAyamma_shrI-vasanta-tyAgarAja.mp3
836
01-sItamma_mAyamma-vasantA.mp3
837
02-Mal_maruga_shanmukha-vasanta-papanasham_shivan.mp3
8

###Updated dictionary with split counts

In [ ]:
resultsDic = {'malavi': 728, 'kuntalavarali': 621, 'natakapriya': 119, 'carukesi': 2551, 'kurinji': 664, 'sindubhairavi': 831, 'hanumatodi': 213, 'bhupalam': 1016, 'andolika': 1358, 'cakravakam': 2164, 'nattakurinji': 3000, 'malayamarutam': 1486, 'ritigaula': 2653, 'madhuvanti': 549, 'sama': 1186, 'kalyani': 14855, 'misrashivaranjani': 594, 'narayanagaula': 1151, 'harikambhoji': 2440, 'revagupti': 736, 'ghanta': 889, 'hindolam': 2968, 'saurashtram': 1277, 'gaulipantu': 723, 'nayaki': 928, 'bauli': 728, 'garudadhvani': 621, 'hamsanadam': 1036, 'athana': 1188, 'hamirkalyani': 1748, 'manji': 1261, 'gambhiranatai': 815, 'darbar': 1327, 'dvijavanti': 1162, 'paras': 606, 'saurastra': 22, 'janaranjani': 914, 'kathanakutuhalam': 0, 'abheri': 4450, 'candrajyoti': 1273, 'kamavardhani': 115, 'lalita': 1157, 'cenjuruti': 583, 'behag': 1910, 'paraju': 554, 'bhairavi': 9037, 'karaharapriya': 2580, 'natabhairavi': 55, 'nariritigaula': 0, 'shivaranjani': 179, 'dhanyasi': 2324, 'kalyanavasantam': 1568, 'shuddhasaveri': 1667, 'bilahari': 3455, 'vasanta': 719, 'varali': 3456, 'saramati': 0, 'saveri': 3329, 'bindumalini': 1035, 'hamsadhvani': 2977, 'jonpuri': 566, 'nalinakanti': 695, 'jaganmohini': 0, 'kamavardhini': 3730, 'shanmukhapriya': 7557, 'nattai': 1317, 'kamboji': 9797, 'mayamalavagaula': 1674, 'mohanakalyani': 1762, 'bahudari': 1256, 'kiravani': 4097, 'madhyamavati': 5206, 'latangi': 2805, 'devamanohari': 1027, 'nagasvaravali': 1199, 'devagandhari': 1296, 'begada': 3171, 'hemavati': 3353, 'manirangu': 1397, 'purnacandrika': 991, 'bowli': 413, 'kedaragaula': 1703, 'kedaram': 1511, 'kapinarayani': 0, 'kambhoji': 108, 'brundavani': 329, 'mishrashivaranjani': 49, 'gaula': 1495, 'gaudamalhar': 677, 'kannadagaula': 947, 'punnagavarali': 619, 'pashupatipriya': 0, 'desh': 533, 'simhendramadhyamam': 3038, 'sourashtram': 170, 'mand': 577, 'navaroj': 465, 'hamsagamini': 0, 'hamsanandi': 1552, 'kanada': 1595, 'anandabhairavi': 3138, 'yamunakalyani': 850, 'karnaranjani': 1265, 'surati': 960, 'huseni': 928, 'shubhapantuvarali': 41, 'vacaspati': 1773, 'saranga': 1139, 'ravicandrika': 725, 'shuddhadhanyasi': 1448, 'shankarabharanam': 8402, 'brundavanasaranga': 1613, 'todi': 14545, 'balahamsa': 640, 'mukhari': 1620, 'nadanamakriya': 881, 'valaji': 971, 'arabhi': 2067, 'nata': 1410, 'salagabhairavi': 733, 'revati': 1193, 'ahiri': 951, 'malahari': 0, 'yadukulakambhoji': 1269, 'khamas': 1465, 'amrutavarsini': 993, 'purvikalyani': 5035, 'pantuvarali': 2743, 'kamalamanohari': 1048, 'kharaharapriya': 8937, 'kannada': 830, 'ranjani': 2701, 'mohanam': 5624, 'sahana': 1399, 'asaveri': 1145, 'nilambari': 572, 'kapi': 2758, 'shriranjani': 2451}

In [ ]:
sum(resultsDic.values())

254491

In [ ]:
totalNumSplits = 0
for ragam in ragamDict:
    totalNumSplits += ragamDict[ragam]
print(totalNumSplits)

261341


In [ ]:
splits_directory_path = '/content/drive/MyDrive/N240Project/RagamSplits3'
ragams140 = os.listdir(splits_directory_path)

In [ ]:
ragamDictLower = {}
for key in ragamDict:
    ragamDictLower[key.lower()] = ragamDict[key]
ragamDictLower

{'ahiri': 810,
 'abheri': 4450,
 'amrutavarsini': 993,
 'anandabhairavi': 3111,
 'andolika': 1358,
 'arabhi': 2067,
 'asaveri': 0,
 'athana': 939,
 'bahudari': 1256,
 'balahamsa': 640,
 'bauli': 728,
 'begada': 3171,
 'behag': 1973,
 'bhairavi': 8650,
 'bhupalam': 1016,
 'bilahari': 3053,
 'bindumalini': 1035,
 'bowli': 376,
 'brundavanasaranga': 1613,
 'brundavani': 329,
 'cakravakam': 2164,
 'candrajyoti': 1273,
 'carukesi': 2551,
 'cenjuruti': 583,
 'darbar': 1327,
 'desh': 943,
 'devagandhari': 1296,
 'devamanohari': 1027,
 'dhanyasi': 2253,
 'dvijavanti': 939,
 'gambhiranatai': 815,
 'garudadhvani': 621,
 'gaudamalhar': 677,
 'gaula': 141,
 'gaulipantu': 723,
 'ghanta': 889,
 'hamirkalyani': 1748,
 'hamsadhvani': 2977,
 'hamsagamini': 0,
 'hamsanadam': 1057,
 'hamsanandi': 1552,
 'hanumatodi': 213,
 'harikambhoji': 3109,
 'hemavati': 3353,
 'hindolam': 1860,
 'huseni': 928,
 'jaganmohini': 0,
 'janaranjani': 948,
 'jonpuri': 566,
 'kalyanavasantam': 1515,
 'kalyani': 14653,
 'kama

In [ ]:
for ragam in ragams140:
    ragamSplitsPath = splits_directory_path + "/" + ragam
    ragamSplits = os.listdir(ragamSplitsPath)
    print(ragam, resultsDic[ragam], len(ragamSplits))

malavi 728 728
kuntalavarali 621 591
natakapriya 119 119
carukesi 2551 2551
kurinji 664 664
sindubhairavi 831 831
hanumatodi 213 213
bhupalam 1016 198
andolika 1358 1358
cakravakam 2164 1357
nattakurinji 3000 3000
malayamarutam 1486 1486
ritigaula 2653 2653
madhuvanti 549 549
sama 1186 1186
kalyani 14855 14855
misrashivaranjani 594 594
narayanagaula 1151 1151
harikambhoji 2440 2440
revagupti 736 736
ghanta 889 889
hindolam 2968 2926
saurashtram 1277 1277
gaulipantu 723 723
nayaki 928 928
bauli 728 728
garudadhvani 621 621
hamsanadam 1036 1036
athana 1188 1188
hamirkalyani 1748 1748
manji 1261 1261
gambhiranatai 815 815
darbar 1327 1327
dvijavanti 1162 1162
paras 606 606
saurastra 22 22
janaranjani 914 914
kathanakutuhalam 0 0
abheri 4450 4436
candrajyoti 1273 1273
kamavardhani 115 115
lalita 1157 1157
cenjuruti 583 583
behag 1910 1910
paraju 554 554
bhairavi 9037 9037
karaharapriya 2580 2580
natabhairavi 55 55
nariritigaula 0 0
shivaranjani 179 179
dhanyasi 2324 2324
kalyanavasantam 15

### Play an example split

In [ ]:
uploaded_file_name = '/content/drive/MyDrive/N240Project/RagamSplits1/bauli/bauLi_split_9'
converted_audio_file = convert_audio_for_model(uploaded_file_name)
sample_rate, data1 = wavfile.read(converted_audio_file, 'rb')
#playsound('converted_audio_file')

Audio(uploaded_file_name)#, autoplay=True)
#print(data1)

